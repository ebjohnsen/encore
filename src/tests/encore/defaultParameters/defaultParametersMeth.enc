active class Main
  def main() : unit
    new Tester()
  end
end

active class Tester
  def init() : unit
    this.foo(1)
    this.foo(2, "hello")
    this.foo(3, "world", 4)
  end

  def foo(a: int, b:String = "b", c:int = 0) : unit
    println(a)
    println(b)
    println(c)
  end
end
