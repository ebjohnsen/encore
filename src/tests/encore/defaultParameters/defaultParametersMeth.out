1
b
0
2
hello
0
3
world
4
